boards/imix/src/main.rs

```diff
@@ -477,16 +490,28 @@ pub unsafe fn reset_handler() {
     rf233_spi.set_client(rf233);
     rf233.initialize(&mut RF233_BUF, &mut RF233_REG_WRITE, &mut RF233_REG_READ);
 
+    let aes_ccm = static_init!(
+        capsules::aes_ccm::AES128CCM<'static, sam4l::aes::Aes<'static>>,
+        capsules::aes_ccm::AES128CCM::new(&sam4l::aes::AES, &mut CRYPT_BUF)
+    );
+    sam4l::aes::AES.set_client(aes_ccm);
+    sam4l::aes::AES.enable();
+
     // Keeps the radio on permanently; pass-through layer
     let awake_mac: &AwakeMac<RF233Device> =
         static_init!(AwakeMac<'static, RF233Device>, AwakeMac::new(rf233));
     rf233.set_transmit_client(awake_mac);
     rf233.set_receive_client(awake_mac, &mut RF233_RX_BUF);
 
     let mac_device = static_init!(
-        capsules::ieee802154::framer::Framer<'static, AwakeMac<'static, RF233Device>>,
-        capsules::ieee802154::framer::Framer::new(awake_mac)
+        capsules::ieee802154::framer::Framer<
+            'static,
+            AwakeMac<'static, RF233Device>,
+            capsules::aes_ccm::AES128CCM<'static, sam4l::aes::Aes<'static>>,
+        >,
+        capsules::ieee802154::framer::Framer::new(awake_mac, aes_ccm)
     );
+    aes_ccm.set_client(mac_device);
```

Comment:
Just to clarify: There shouldn't be any problem calling this function from a "Thread" capsule to set the client as the Thread control plane for instances when MLE messages are being sent and the link layer is not encrypting messages, correct? It would obviously need a reference to the mac_device as well so that it could set the client back once the MLE process was complete. I could see this being an issue in cases where one app is running Thread and another is simply trying to communicate using encrypted 15.4 - somehow there would need to be a mechanism to ensure that the mac_device is always set as the client when link-layer encryption is needed, while also allowing Thread to be set as the client when above-link-layer encryption is needed for MLE messages.
That seems like the only potential issue which could hinder the 6lowpan/Thread development down the road, and even if it is something that will eventually have to change I think the current implementation is fine and will not hinder the current 6lowpan/Thread development as I expect getting CCM* working on MLE messages will be one of the final steps towards a SED